mvrx/src/main/kotlin/com/airbnb/mvrx/CoroutinesStateStore.kt

```diff
@@ -0,0 +1,128 @@
+package com.airbnb.mvrx
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.channels.BroadcastChannel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.consumeEach
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.channelFlow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import java.util.concurrent.Executors
+
+@Suppress("EXPERIMENTAL_API_USAGE")
+class CoroutinesStateStore<S : MvRxState>(
+    initialState: S,
+    scope: CoroutineScope
+) : MvRxStateStore<S> {
+
+    /** Channel that serves as a trigger to flush the setState and withState queues. */
+    private val flushQueuesChannel = Channel<Unit>(capacity = Channel.CONFLATED)
+    private val setStateChannel = Channel<S.() -> S>(capacity = Channel.UNLIMITED)
+    private val withStateChannel = Channel<(S) -> Unit>(capacity = Channel.UNLIMITED)
+
+    /**
+     * This combination of BroadcastChannel, mutable state property and channelFlow was arrived
+     * at after multiple attempts. This code needs to:
+     * 1) Multicast
+     * 2) Never fail to deliver an intermediate state to subscribers even if updates are fast or
+     *    some subscribers are slow.
+     * 3) Be able to provide an initial value.
+     *
+     * StateFlow can't be used because it conflates new sets. If you set its value in a tight
+     * loop and listen to changes on a flow, you will miss values. ConflatedBroadcastChannel
+     * alone has the same issue.
+     *
+     * A normal Channel can't be used because it isn't multicast.
+     */
+    private val stateChannel = BroadcastChannel<S>(capacity = Channel.BUFFERED)
+    override var state = initialState
+    override val flow: Flow<S> get() = channelFlow {
+        send(state)
+        stateChannel.consumeEach { send(it) }
+    }.distinctUntilChanged()
```

Comment:
If you remove `buffer()` from BaseMvRxViewModel.kt according my comment, you probably need to add buffer(1) here before disticntUntilChanged to consume `send(state)` and subscribe to stateChannel immediately.